### PR TITLE
nano: add without-nls option.

### PR DESCRIPTION
**Pull request by arabesc**

### CODE REVIEW

**Comment by dunn** [comment]
Should this option remove the hard dependency on gettext?

**Comment by bfontaine** [comment]
Ping on Alex’s comment above?

**Comment by arabesc** [comment]
It's ready.

File: nano.rb [hunk 1]
```diff
@@ -17,8 +17,10 @@ class Nano < Formula
     depends_on "autoconf" => :build
   end
 
+  option "without-nls", "Build without Native Language Support"
+
   depends_on "pkg-config" => :build
-  depends_on "gettext"
+  depends_on "gettext" if build.with? "nls"
```
Comment (ilovezfs) [inline nano.rb]:
```
depends_on "gettext" => :recommended
```

File: nano.rb [hunk 2]
```diff
@@ -17,8 +17,10 @@ class Nano < Formula
     depends_on "autoconf" => :build
   end
 
+  option "without-nls", "Build without Native Language Support"
```
Comment (ilovezfs) [inline nano.rb]:
```
option "without-gettext", "Build without Native Language Support"
```

File: nano.rb (hunk 1 above)
Comment (arabesc) [inline nano.rb]: Why is it better than `if build.with? "nls"`?

Comment (ilovezfs) [inline nano.rb]: because it's visible to the dependency system as it's bound to a formula

File: nano.rb (hunk 2 above)
Comment (arabesc) [inline nano.rb]: Do you suggest to replace "without-nls" to "without-gettext" or to add a new option?

Comment (arabesc) [inline nano.rb]: I think that "without-nls" is more widespread and obvious.

Comment (ilovezfs) [inline nano.rb]: It has to match the `=> :recommended`, which creates the option. This just overrides the default description.

Comment (arabesc) [inline nano.rb]: I see. I have no expertise in the area to decide, so I suggest someone clone the changes and do it further in a right way.

Comment (ilovezfs) [inline nano.rb]: @arabesc I've updated the PR including fixing the head spec.

**Comment by MikeMcQuaid** [comment]
This formula has been moved to homebrew/core. If @ilovezfs is cool on accepting it here it'll need to be moved over. Sorry!

**Pull request closed without merging**